src/transformStateWithClones.js

```diff
@@ -8,6 +8,33 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const resultState = [];
+  let tempState = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(tempState, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete tempState[key];
+        }
+        break;
+
+      case 'clear':
+        tempState = {};
+        break;
+    }
+
+    let newState = { ...tempState };
+
+    resultState.push(newState);
+    newState = {};
```

Comment:
The line `newState = {};` is unnecessary because `newState` is redefined in each iteration of the loop. Removing this line will not affect the functionality of the code.